src/features/cleaner_file_duplicate.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::{Context, Result};
+use colored::*;
+use std::collections::HashSet;
+use std::fs;
+use walkdir::WalkDir;
+
+fn get_file_key(file_name: Option<&std::ffi::OsStr>, file_size: u64) -> Option<String> {
+    file_name.and_then(|name| name.to_str()).map(|name| {
+        // Extract the first part of the file name, before any whitespace or special characters
+        let first_part = name.split_whitespace().next().unwrap_or("");
+        format!("{}{}", first_part, file_size)
+    })
+}
+
+pub fn directory_cleaner_based_on_duplicate_files(directory: &String, dry_run: bool) -> Result<()> {
+    let mut set: HashSet<String> = HashSet::new();
+    for entry in WalkDir::new(directory).into_iter() {
+        match entry {
+            Ok(dir) => {
+                let path = dir.path();
+
+                if path.is_file() {
+                    let metadata = fs::metadata(path)
```

Comment:
same here, maybe create an util class, and we can just do getMetadata. etc

Reply:
so getMetadata would be `fs::metadata(path).with_context(|| format!("Failed to read metadata for file: {:?}", path))?;`. Doesn't make it cleaner IMO. That's a small enough code blob